### PR TITLE
Add alt text to JupyterLab documentation

### PR DESCRIPTION
**Pull request by isabela-pf**
# Welcome to the working PR!

This is the working PR for the SciPy sprint alt text workshops! This is where we collaborate during the event before we send a clean contribution over to the [main repo](https://github.com/jupyterlab/jupyterlab).

For the full info about this event, please visit the [event agenda] (to be linked).

## How to contribute

⭐ For full info about how to interact with this PR, please visit the [How to Participate section of the agenda] (to be linked). Please contribute via [suggestions](https://hackmd.io/@marslee/BksnYfWVK) so we can easily review each other's work and give you credit in the commit history!

✏️ If you plan to work on a file, let people know by commenting below. You can do this as a group, and you can write alt text for as many images as you'd like. We can have multiple suggestions for the same alt text, but it would be nice if we spread out to cover more images.

📚 If you need a reference for how to format your suggestions, [alt text in Markdown files (ends with `.md`) follow this format](https://www.markdownguide.org/basic-syntax/#images-1) while [alt text in RST files (ends with `.rst`) follow this format](https://sublime-and-sphinx-guide.readthedocs.io/en/latest/images.html).

🖼️ There are two types of images in this PR: ones that already have alt text and ones that have none. 
- Images that appear in this PR with existing alt text are still worth reviewing and editing! If you think they could benefit from a different description, please suggest one.
- Images with no alt text need a description added in place of the blank.

❗ If you think an image is missing from these files, let @isabela-pf know and they will add it to the PR.

## Images in this PR

---

If you need anything, please ask @isabela-pf 🌻

### CODE REVIEW

File: docs/source/developer/contributing.rst [hunk 1]
```diff
@@ -787,6 +790,7 @@ The first step is to create a test file and a debug configuration (``launch.json
 Then start the debugger:
 
 .. image:: ./debugger_vscode_start.png
+   :alt:
```
Comment (aerikpawson) [inline docs/source/developer/contributing.rst]:
```suggestion
   :alt: A screenshot of code editor with test.py open, staring the debugger.
```

File: docs/source/developer/internationalization.rst [hunk 2]
```diff
@@ -102,6 +102,7 @@ There is one optional setting:
 - The new version in form *X.Y.postZ* - if not provided, the post number will be bumped.
 
 .. image:: prep_language_packs.png
+  :alt:
```
Comment (aerikpawson) [inline docs/source/developer/internationalization.rst]:
```suggestion
  :alt: A screenshot showcasing the "Prepare language packs for release" workflow on GitHub Actions.
```

File: docs/source/user/debugger.rst [hunk 3]
```diff
@@ -38,6 +38,7 @@ Usage
 Here is a screencast to enable the debugger and set up breakpoints. The various steps are described more in depth below.
 
 .. image:: ./images/debugger/step.gif
+   :alt:
```
Comment (aerikpawson) [inline docs/source/user/debugger.rst]: A .gif file showing how to enable the debugger within the kernel.

File: docs/source/user/toc.rst [hunk 4]
```diff
@@ -14,6 +14,7 @@ In the sidebar panel, you can number headings, collapse sections, and navigate i
 Here is an animation showing the table of content use with a notebook:
 
 .. image:: ./images/toc/toc.gif
+    :alt:
```
Comment (aerikpawson) [inline docs/source/user/toc.rst]: A .gif file displaying how to use the table of contents feature within a notebook.

File: docs/source/user/toc.rst [hunk 5]
```diff
@@ -36,6 +37,7 @@ option via the *More actions* button in the table of contents toolbar. Here is a
 showing its use:
 
 .. image:: ./images/toc/numberingH1.gif
+    :alt:
```
Comment (aerikpawson) [inline docs/source/user/toc.rst]:
```suggestion
    :alt:A .gif file showing how to toggle the "Automatic section numbering" feature for the first-level headings (h1) within a notebook.
```

File: docs/source/user/debugger.rst (hunk 3 above)
Comment (isabela-pf) [reply docs/source/user/debugger.rst]: Hey there, thanks for contributing! If you'd be able to edit your above comment into a [GitHub suggestion](https://hackmd.io/@marslee/BksnYfWVK), I'd appreciate so I can make sure you get credit in the commit history. If not, I'll still add it with an extended description that mentions you.

File: docs/source/user/toc.rst (hunk 4 above)
Comment (isabela-pf) [reply docs/source/user/toc.rst]: Hey there, thanks for contributing! If you'd be able to edit your above comment into a [GitHub suggestion](https://hackmd.io/@marslee/BksnYfWVK), I'd appreciate so I can make sure you get credit in the commit history. If not, I'll still add it with an extended description that mentions you.

File: docs/source/user/debugger.rst (hunk 3 above)
Comment (isabela-pf) [reply docs/source/user/debugger.rst]:
```suggestion
   :alt: A .gif file showing how to enable the debugger within the kernel.
```

File: docs/source/user/toc.rst (hunk 4 above)
Comment (isabela-pf) [reply docs/source/user/toc.rst]:
```suggestion
    :alt: A .gif file displaying how to use the table of contents feature within a notebook.
```

File: docs/source/user/debugger.rst (hunk 3 above)
Comment (aerikpawson) [reply docs/source/user/debugger.rst]: @isabela-pf My apologies for the slow response and thank you for fixing my mishaps.

Comment (isabela-pf) [reply docs/source/user/debugger.rst]: No apology necessary! Thank *you* for writing alt text and improving our docs!

**Pull request merged**